sql/src/main/java/io/crate/execution/dsl/phases/HashJoinPhase.java

```diff
@@ -76,6 +76,7 @@ public HashJoinPhase(UUID jobId,
             JoinType.INNER,
             joinCondition);
         assert joinCondition != null : "JoinCondition for HashJoin cannot be null";
+        assert rowsToBeConsumed > 0 : "RowsToBeConsumed for HashJoin must be greater than 0";
```

Comment:
strictly saying, 0 is valid, only negative not (indeed, limit 0 would be optimized to no match earlier)